pilot/pkg/networking/core/v1alpha3/envoyfilter/rc_patch.go

```diff
@@ -251,7 +256,7 @@ func patchHTTPRoute(patchContext networking.EnvoyFilter_PatchContext,
 				*routesRemoved = true
 				return
 			} else if rp.Operation == networking.EnvoyFilter_Patch_MERGE {
-				proto.Merge(virtualHost.Routes[routeIndex], rp.Value)
+				safeMerge(virtualHost.Routes[routeIndex].Name, virtualHost.Routes[routeIndex], rp.Value)
```

Comment:
`safeMerge` is not recursively checking the `name` to see if its safe, so don't we end up with an issue if we merge at a high level? For example:

```
knownCatchedObjects := ["foo"]
x := {bar:{foo:{}}}
safeMerge(x, ...)
```

We will check name (bar), but then merge and foo will get overwriten